include/libtorrent/torrent.hpp

```diff
@@ -1481,7 +1481,7 @@ namespace libtorrent
 // ----
 
 		// the maximum number of uploads for this torrent
-		unsigned int m_max_uploads:24;
+		std::uint32_t m_max_uploads:24;
```

Comment:
I made some change, but maybe you want to take a look at the entire layout again. I have the impression that maybe something is not up to date